### PR TITLE
feat: set up GH Actions frontend

### LINKED ISSUE #8: Set up GH Actions
As a service, 
I want to have GH Actions, 
so that I can automatically trigger appropriate actions.

### Solution notes
- store secrets (DB password / API keys / etc.) in `GitHub Actions Secrets`
- runs-on: ubuntu-latest

Actions:
- deploy application on main branch update (if applicable)
- run lint and tests on PR creation/update
- (optional) rebase to main branch

### Resources:

* https://docs.github.com/en/actions

### PR DESCRIPTION
**Pull request by OlyaPolya**
### Task closes #8 
* [ ] `npm run lint` check passed without any failures or warnings
* [ ] functionality is covered by new tests
* [ ] functionality works in production build
* [ ] existing functionality remains unchanged
* [ ] updated documentation (if required)

### CODE REVIEW

File: .github/workflows/deploy-to-Netlify.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: Deploy site to Netlify
```
Comment (AnhelinaVr) [inline .github/workflows/deploy-to-Netlify.yml]: pls rename to `deploy.sh`
we don’t have to specify deploy platform as we have only one option

Comment (AnhelinaVr) [reply .github/workflows/deploy-to-Netlify.yml]:
```suggestion
name: Deploy application
```

File: .github/workflows/lint-and-tests.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: Run lint and tests on PR
+on:
+  pull_request:
+      types: [opened, edited]
+      branches:
+      - main
```
Comment (AnhelinaVr) [inline .github/workflows/lint-and-tests.yml]: will the PR created to other branches trigger this hook?

File: .github/workflows/lint-and-tests.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+jobs:
+  run-lint-and-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+      - run: npm ci
+      - name: Tests application
```
Comment (AnhelinaVr) [inline .github/workflows/lint-and-tests.yml]:
```suggestion
      - name: Run tests
```

File: .github/workflows/lint-and-tests.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        run: npm run test
+      - name: Lint application
```
Comment (AnhelinaVr) [inline .github/workflows/lint-and-tests.yml]:
```suggestion
      - name: Run lint
```

File: .github/workflows/deploy-to-Netlify.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+on:
+  push:
+      branches:
+      - main
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Deploy site
```
Comment (AnhelinaVr) [inline .github/workflows/deploy-to-Netlify.yml]:
```suggestion
      - name: Deploy application
```

File: .github/workflows/deploy-to-Netlify.yml (hunk 1 above)
Comment (OlyaPolya) [reply .github/workflows/deploy-to-Netlify.yml]: Done

File: .github/workflows/lint-and-tests.yml (hunk 2 above)
Comment (OlyaPolya) [reply .github/workflows/lint-and-tests.yml]: No, I made it only for main branch. Should it runs for other branches?

File: .github/workflows/lint-and-tests.yml (hunk 3 above)
Comment (OlyaPolya) [reply .github/workflows/lint-and-tests.yml]: Done

File: .github/workflows/lint-and-tests.yml (hunk 4 above)
Comment (OlyaPolya) [reply .github/workflows/lint-and-tests.yml]: Done

File: .github/workflows/deploy-to-Netlify.yml (hunk 5 above)
Comment (OlyaPolya) [reply .github/workflows/deploy-to-Netlify.yml]: Done

File: .github/workflows/lint-and-tests.yml (hunk 2 above)
Comment (AnhelinaVr) [reply .github/workflows/lint-and-tests.yml]: I suppose yes, 'cause ensuring the quality of the code, regardless of which branch to merge it into, is a rule of thumb

File: .github/workflows/lint-and-tests.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        with:
+          node-version: '16'
+      - run: npm ci
+      - name: Run tests
+        run: npm run test
+      - name: Run lint
+        run: npm run lint
+
+
+
```
Comment (AnhelinaVr) [inline .github/workflows/lint-and-tests.yml]: also please remove redundant empty lines

File: .github/workflows/lint-and-tests.yml (hunk 2 above)
Comment (OlyaPolya) [reply .github/workflows/lint-and-tests.yml]: Done

File: .github/workflows/lint-and-tests.yml (hunk 6 above)
Comment (OlyaPolya) [reply .github/workflows/lint-and-tests.yml]: Done

**Review by AnhelinaVr — APPROVED** [review]

**Review by Konstantsin-Shumak — APPROVED** [review]

**Review by AnhelinaVr — APPROVED** [review]

**Pull request merged**